### PR TITLE
is1047/prepare-qx-kit

### PR DESCRIPTION
**Pull request by pcrespov**
Prepares toolkit to remove webclient service from simcore stack as part of case ITISFoundation/osparc-simcore#1047

- [x] update Dockerfile to latest version from webclient in osparc-simcore
- [x] implements makefile best-practices: ``make help info``, ``make build-nc test release`` or ``make shell``
- [x] sample integration 
- [x] review related cases and close
-  ~review tag, release cycle~
-  ~add CI in travis to test and deploy~  will do in #16

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,69 +1,37 @@
...
+build-nc: ## Builds all images from scratch
+	docker-compose build --no-cache
 
 .PHONY: deploy
-# target: deploy – Tags images and pushes to dockerhub's registry [TODO]
-deploy: .env 
+publish:  ## TODO: Tags images and pushes to dockerhub's registry
 	echo TODO: tagging images and pushing to dockerhub ...
-	#${DOCKER_COMPOSE} push
+	#docker-compose push
```
Comment (oetiker) [inline Makefile]: Of late I started to add a safeguard to make sure one could only tag an image if on origin/master ... 

```
uptodate-test: ## Have all the changes been commited ?
        @ git fetch upstream
        @if [ "$$(git diff upstream/master)" != "" ]; then \
                echo "*** Can't tag/release/publish if you are not in sync with upstream/master ***"; \
                exit 1; \
        fi
```

and then protext the tag target in this way

```
tag-latest: uptodate-test ## Tag easii-prod:latest container
        @echo 'create tag latest'
        docker tag $(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
```

File: qooxdoo-kit/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,31 @@
-FROM node:10-alpine
+FROM node:10.16.3-alpine
```
Comment (oetiker) [inline qooxdoo-kit/Dockerfile]: maybe have the version in an ARG with a default to make it more explicit

```
ARG ALPINE_VERSION=10.16.3
FROM node:${ALPINE_VERSION}-alpine
```

Comment (pcrespov) [reply qooxdoo-kit/Dockerfile]: I see your point but I think this is overdoing a bit at this stage :-)

I find this useful to e.g. sync versions among base images within this repo or in a multistage build as explained in
https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact.

Comment (oetiker) [reply qooxdoo-kit/Dockerfile]: whatever you mean by this ... the idea is about 'readable code' and implicit documentation ... as a casual reader might not get that this is the node version and not the version of alpine ... just an idea ...

Comment (pcrespov) [reply qooxdoo-kit/Dockerfile]: As I said, I understand your point and I take note. thx :-)

File: Makefile [hunk 3]
```diff
@@ -1,69 +1,37 @@
...
-# target: help – Display all callable targets
-help:
-	@echo
-	@egrep "^\s*#\s*target\s*:\s*" [Mm]akefile \
-	| $(SED) -r "s/^\s*#\s*target\s*:\s*//g"
-	@echo
\ No newline at end of file
+.PHONY: clean
+clean: ## Cleans all unversioned files in project
+	@git clean -dxf
```
Comment (mguidon) [inline Makefile]: After I once lost a full afternoon of work I started adding
```bash
clean: check_clean
    @git clean -dxf 
check_clean:
    @echo -n "Are you sure? [y/N] " && read ans && [ $${ans:-N} = y ]
```

File: Makefile (hunk 1 above)
Comment (pcrespov) [reply Makefile]: will consider in #16

**Pull request merged**